src/averageAges.js

```diff
@@ -55,10 +69,30 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people
+    .filter((person) => (
+      person.mother !== null
+      && people.find((mother) => mother.name === person.mother)
+    ))
+    .filter((person) => (
+      onlyWithSon === undefined
+      || person.sex === 'm'
```

Comment:
* Consider creating variables with meaningful names instead of using magic strings
* Follow the DRY principle: don't repeat yourself. You are using the sex check a lot of times. Consider creating a function for this purpose